src/dialog/an-dialog.factory.js

```diff
@@ -151,6 +160,53 @@
                         });
                     };
 
+                    var getFocusableElements = function() {
+                        var focusableEls = element.find('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
+
+                        focusableElems = Array.prototype.slice.call(focusableEls);
+                    };
+
+                    var setInitialFocusableElements = function () {
+                        firstFocusableElem = focusableElems[0];
+                        lastFocusableElem = focusableElems[focusableElems.length - 1];
+                    };
+
+                    var onDialogKeyDownEvent = function(e) {
+                        var KEY_TAB = 9;
+
+                        switch(e.keyCode) {
+                            case KEY_TAB:
+                                if (firstFocusableElem.length === 1) {
```

Comment:
Shouldn't this be checking for focusableElems.length instead?

Reply:
done